### PR TITLE
spack recipe for gromacs with aocc compiler support

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
Recipe for gromacs with aocc compiler support.

### CODE REVIEW

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 1]
```diff
@@ -234,5 +248,26 @@ def cmake_args(self):
...
+                options.append(
+                    '-DFFTWF_INCLUDE_DIRS={0}'.
+                    format(self.spec['amdfftw'].headers.directories[0])
+                )
+                options.append('-DFFTWF_LIBRARIES={0}'.
+                               format(self.spec['amdfftw'].libs.joined(';')))
+
+        if '%aocc' in self.spec:
+            options.append('-DGMX_CYCLE_SUBCOUNTERS:BOOL=OFF')
+            options.append('-DCMAKE_C_FLAGS=-Ofast -march=znver2 -ffast-math')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gromacs/package.py]: Isn't this `-march` flag machine specific? Spack already injects optimization flags automatically. I don't think we have `%aocc` support yet in `archspec`, but it shouldn't be hard to add. @alalazo

**Review by adamjstewart — CHANGES_REQUESTED** [review]
@junghans @marvinbernhardt can you review?

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 2]
```diff
@@ -88,15 +94,17 @@ class Gromacs(CMakePackage):
     depends_on('plumed+mpi', when='+plumed+mpi')
     depends_on('plumed~mpi', when='+plumed~mpi')
     depends_on('fftw-api@3')
+    depends_on('mkl', when='fft=mkl')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/gromacs/package.py:97]: @amd-toolchain-support The new concretizer errors on the fact that `gromacs` has no `fft` variant.

Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/gromacs/package.py:97]: @alalazo we were referring old change when we raised a pull request. It has been fixed

**Pull request merged**